yopt/src/main/scala/pravda/yopt/CmdDecoder.scala

```diff
@@ -53,22 +57,94 @@ object CmdDecoder {
     override def decode(line: Line): Either[String, (Long, Line)] =
       line.headOption
         .map { item =>
-          try {
-            Right((java.lang.Long.parseLong(item), line.tail))
-          } catch {
-            case ex: NumberFormatException =>
-              try {
-                Right((java.lang.Long.parseLong(item.drop(2), 16), line.tail))
-              } catch {
-                case ex: NumberFormatException => parseError(ex)
-              }
+          item.toLowerCase match {
+            case neg if neg.head == '-' => Left(NegativeNumberDecodeError.toString)
+            case hex if hex.startsWith("0x") =>
+              Try(java.lang.Long.parseLong(hex.drop(2), 16)).fold(
+                _ => Left(LongDecodeError(hex).toString),
+                number => Right( (number, line.tail) )
+              )
+            case moreLikelyDec => Try(moreLikelyDec.toLong).fold(
+              _ => Left(LongDecodeError(moreLikelyDec).toString),
+              number => Right( (number, line.tail) )
+            )
           }
         }
         .getOrElse(noValueError)
 
     override val optInfo: Option[String] = Some("<long>")
   }
 
+  implicit val booleanReader: CmdDecoder[Boolean] = new CmdDecoder[Boolean] {
+    override def decode(line: Line): Either[String, (Boolean, Line)] =
+      line.headOption
+        .map { item =>
+          item.toLowerCase match {
+            case "true"  => Right( (true, line.tail) )
```

Comment:
There's the same problem. Why do you think user always understands that `Boolean` parser is able to parse `y` as `true`? For example we could have some boolean options `-y` and `-n`, so with your parser `some-command -y n -n y` expression is correct.

Reply:
Yes, to rid of ambiguity I will remove "y" and "n" letters as correct options for a boolean value.